### PR TITLE
Issue 17 dev environment contrib standards

### LINKED ISSUE #17: Instructions for setting up a dev environment and contribution standards
**Is your feature request related to a problem? Please describe.**
Contributing this this project is currently slowed down by not having instructions for a standardized dev environment set up. We also observed that PR contributions are not formatted consistently.

**Describe the solution you'd like**
1. Clear instructions and supporting file for setting up a standardized dev environment. 
2. Instructions for setting up relevant pre-commit hooks

**Describe alternatives you've considered**
NA

**Additional context**
NA

### PR DESCRIPTION
**Pull request by iamchrisearle**
## Description

Closes #17 

- ~~Adds `pyproject.toml` and `poetry.lock` for virtual environment setup using poetry~~
- Adds [pre-commit hooks using ruff ](https://github.com/astral-sh/ruff-pre-commit)
- Several minor file changes from running ruff formatter
- Adds relevant documentation to readme

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

- [x] I have run pytest

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Next Steps
- [x] Assign a reviewer based on the [code owner document](https://github.com/fraymio/modis_tools/blob/main/.github/CODEOWNERS).

- [ ] Once your review is approved, merge and delete the feature branch

On behalf of the Modis Tools Dev Team, thank you for your hard work! ✨

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "modis-tools"
+version = "0.1.0"
+description = "An enivornment for modis tools to download MODIS imagery from the NASA Earthdata platform"
+authors = ["Chris <chris.c.earle@gmail.com>"]
```
Comment (jamie-sgro) [inline pyproject.toml:9]: How bold Chris :P

File: modis_tools/auth.py [hunk 2]
```diff
@@ -36,7 +35,8 @@ def __init__(
                 username, _, password = netrc().authenticators(URLs.URS.value)
             except FileNotFoundError:
                 raise Exception(
-                    "Unable to create authenticated session. Likely that username and password not found"
+                    "Unable to create authenticated session.\
+                        Likely that username and password not found"
```
Comment (jamie-sgro) [inline modis_tools/auth.py:39]: From https://app.clickup.com/36149938/v/dc/12f6nj-75253/12f6nj-139813 under formatting

> dont_do_this = "This is a very very very very \
                  long string"
# Outputs: "This is a very very very very long string"
do_this_instead = (
    "This is a very very "
    "long string"
)
# Outputs: "This is a very very long string"

File: tests/test_models.py
Comment (jamie-sgro) [inline tests/test_models.py:1]: I'll need a comment from you as to what this change is aiming to accomplish

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "modis-tools"
+version = "0.1.0"
+description = "An enivornment for modis tools to download MODIS imagery from the NASA Earthdata platform"
```
Comment (jamie-sgro) [inline pyproject.toml:8]: enivornment -> environment

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+[build-system]
+requires = ["poetry-core>=1.0.0"]
+build-backend = "poetry.core.masonry.api"
+
+[tool.poetry]
+name = "modis-tools"
+version = "0.1.0"
```
Comment (jamie-sgro) [inline pyproject.toml:7]: These values don't match setup.py

File: pyproject.toml
Comment (jamie-sgro) [inline pyproject.toml:1]: I reckon we should remove this file for the reasons listed in this PR

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+authors = ["Chris <chris.c.earle@gmail.com>"]
+
+[tool.poetry.dependencies]
+python = "^3.7.17" 
+requests = "2.*"
+pydantic = "1.*"
+python-dateutil = "2.*"
+shapely = "2.*"
+tqdm = "^4.42.0"
+setuptools = {version = "69.2.0", markers = "python_version >= '3.12'"}
```
Comment (jamie-sgro) [inline pyproject.toml:18]: This file is missing two extras:

    extras_require={
        "test": ["pytest"],
        "gdal": ["gdal"],
    }

File: README.md [hunk 6]
```diff
@@ -203,14 +203,34 @@ Currently modis_tools only supports downloading of hdf file type.
...
 
-- To install all production dependencies, run:
+- The developers use [poetry](https://python-poetry.org/) to manage virtual environments and dependencies. Included in root is a `pyproject.toml` and `poetry.lock`. To quickly setup a virtual environment locally, from root simply run:
+
+  ```bash
+  poetry shell
+  poetry update
+  ```
+
+- Alternatively, To install all production dependencies, run:
```
Comment (jamie-sgro) [inline README.md:226]: I don't think the move is to add poetry here. We deploy this as a pip installable package using requirements and setup.py. Adding anything related to poetry conflates access strategy. Specifically, the pyproject.toml created here (despite not being specifically a poetry file) will cause drift with setup.py. I recommend we remove the pyproj

**Review by jamie-sgro — CHANGES_REQUESTED** [review]
All looks good except for the pyproj and a string split

**Comment by iamchrisearle** [comment]
> All looks good except for the pyproj and a string split

Removed poetry references -- back to you

**Review by jamie-sgro — APPROVED** [review]
LGTM

**Pull request merged**